### PR TITLE
Fix Clippy Warnings for Leptos 0.7

### PR DESCRIPTION
**Pull request by SleeplessOne1917**
I was taking a look at this project and kept getting a bunch of warnings when I ran `cargo check`. This PR fixes all the warnings from running `cargo check` and `cargo clippy`.

### CODE REVIEW

File: src/use_cycle_list.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ pub fn use_cycle_list<T, L>(
 >
 where
     T: Clone + PartialEq + Send + Sync + 'static,
-    L: Into<MaybeSignal<Vec<T>>>,
+    L: Into<Signal<Vec<T>>>,
```
Comment (SleeplessOne1917) [inline src/use_cycle_list.rs:44]: The warning for uses of `MaybeSignal` was

```
warning: use of deprecated enum `leptos::prelude::MaybeSignal`: `MaybeSignal<T>` is deprecated in favour of `Signal<T>` which is `Copy`, now has a more efficient From<T> implementation and other benefits in 0.7.
```

**Comment by maccesch** [comment]
Thanks you very much!

**Comment by SleeplessOne1917** [comment]
@maccesch Are the failing tests something to be concerned about?

**Comment by maccesch** [comment]
No, don't worry, I wanted to do some touchups anyway 👍🏼

**Pull request merged**